docs/src/tutorial-kinds-of-failure.md

```diff
@@ -63,54 +76,57 @@ Any unsafe code that manipulates pointers will, as we see here, raise failures i
 
 Consider trying a few more small exercises with this example:
 
-1. Exercise: Switch back to the normal/safe indexing operation and re-try Kani. What changes compared to the unsafe operation and why?
+1. Exercise: Switch back to the normal/safe indexing operation and re-try Kani.
+How does Kani's output change, compared to the unsafe operation?
 (Try predicting the answer, then seeing if you got it right.)
 2. Exercise: [Remember how to get a trace from Kani?](./tutorial-first-steps.md#getting-a-trace) Find out what inputs it failed on.
 3. Exercise: Fix the error, run Kani, and see a successful verification.
-4. Exercise: Try switching back to the unsafe code (now with the error fixed) and re-run Kani. It should still successfully verify.
+4. Exercise: Try switching back to the unsafe code (now with the error fixed) and re-run Kani. Does it still verify successfully?
 
 <details>
 <summary>Click to see explanation for exercise 1</summary>
 
 Having switched back to the safe indexing operation, Kani reports two failures:
 
 ```
-# kani src/bounds_check.rs --harness bound_check | grep Failed
+SUMMARY: 
+ ** 2 of 350 failed
 Failed Checks: index out of bounds: the length is less than or equal to the given index
+ File: "./src/bounds_check.rs", line 11, in bounds_check::get_wrapped
 Failed Checks: dereference failure: pointer outside object bounds
+ File: "./src/bounds_check.rs", line 11, in bounds_check::get_wrapped
+
+VERIFICATION:- FAILED
 ```
 
-The first is Rust's implicit assertion for the safe indexing operation.
+The first is Rust's runtime bounds checking for the safe indexing operation.
 The second is Kani's check to ensure the pointer operation is actually safe.
-This pattern (two checks for similar issues in safe Rust code) is common, and we'll see it again in the next section.
+This pattern (two checks for similar issues in safe Rust code) is common to see, and we'll see it again in the next section.
+
+> **NOTE**: While Kani will always emit both kinds of checks, in the future the output here may change.
+> Kani is not currently recognizing that the runtime bounds check means it cannot reach the unsafe pointer dereference.
```

Comment:
Suggest rephrasing to something along the lines of:
Kani does not currently realize that the unsafe pointer dereference check is redundant: if the runtime bounds check passes, the pointer dereference check will also pass, and it the runtime bounds check fails, the pointer dereference check will also fail (or will be unreachable?).

Reply:
Ok, maybe this is our miscommunication here.

It's not redundant checks: I'm trying to say we'd always emit both.

What I suspect we're getting here is that the "assert" that Rust is inserting for the bounds check is getting translated as "assert" to cbmc, not our usual "assert-assume".